File: CaaS.Api/CartApi/Models/CartDto.cs
using System.Collections.Immutable;
using CaaS.Api.CouponApi.Models;
using CaaS.Api.CustomerApi.Models;
using CaaS.Api.DiscountApi.Models;

namespace CaaS.Api.CartApi.Models; 

public record CartDto {
    public Guid Id { get; init; }
    public Guid ShopId { get; init; }
    public CustomerDto? Customer { get; init; } = CustomerDto.Empty;
    public IReadOnlyList<CartItemDto> Items { get; init; } = ImmutableArray<CartItemDto>.Empty;
    public IReadOnlyList<CouponDto> Coupons { get; init; } = ImmutableArray<CouponDto>.Empty;
    public IReadOnlyList<DiscountDto> CartDiscounts { get; init; } = ImmutableArray<DiscountDto>.Empty;
    public DateTimeOffset LastAccess { get; init; }
}

File: src/app/shared/cart/models/cartDto.ts
import { DiscountDto } from './discountDto';
import { CartItemDto } from './cartItemDto';
import { CustomerDto } from './customerDto';
import { CouponDto } from './couponDto';


export interface CartDto {
    id?: string;
    shopId?: string;
    customer?: CustomerDto;
    items?: Array<CartItemDto> | null;
    coupons?: Array<CouponDto> | null;
    cartDiscounts?: Array<DiscountDto> | null;
    lastAccess?: string;
    totalPrice?: number;
    concurrencyToken?: string | null;
}

